File: solutions/fable-1/fable/perl/wordfreq.pl
use strict;
use warnings;

my %count;
while (my $line = <STDIN>) {
    while ($line =~ /([A-Za-z]+)/g) {
        $count{lc $1}++;
    }
}

for my $w (sort { $count{$b} <=> $count{$a} || $a cmp $b } keys %count) {
    print "$w $count{$w}\n";
}

File: solutions/fable-1/fable/rust/wordfreq.rs
use std::collections::HashMap;
use std::io::{self, Read, Write};

fn main() {
    let mut input = Vec::new();
    io::stdin().read_to_end(&mut input).unwrap();

    let mut counts: HashMap<String, u64> = HashMap::new();
    let mut cur = String::new();
    for &b in input.iter() {
        if b.is_ascii_alphabetic() {
            cur.push(b.to_ascii_lowercase() as char);
        } else if !cur.is_empty() {
            *counts.entry(std::mem::take(&mut cur)).or_insert(0) += 1;
        }
    }
    if !cur.is_empty() {
        *counts.entry(cur).or_insert(0) += 1;
    }

    let mut entries: Vec<(String, u64)> = counts.into_iter().collect();
    entries.sort_by(|a, b| b.1.cmp(&a.1).then_with(|| a.0.cmp(&b.0)));

    let stdout = io::stdout();
    let mut out = io::BufWriter::new(stdout.lock());
    for (word, count) in entries {
        writeln!(out, "{} {}", word, count).unwrap();
    }
}
